### PR TITLE
GR: Add tests for npmrc & npm registry api

### PR DESCRIPTION
**Pull request by michaelkedar**
Add some tests for `npmrc` parsing / registry request building and querying of the npm registries.

Currently, the `npmrc` parser can use a few environment variables, which makes it a bit difficult to test properly. I may change it in another PR to make it take them in as an argument so I can sandbox the tests better.

### CODE REVIEW

File: internal/resolution/datasource/npm_registry_test.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"slices"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/google/osv-scanner/internal/resolution/datasource"
+	"github.com/google/osv-scanner/internal/testutility"
+)
+
+type fakeNpmRegistry struct {
```
Comment (michaelkedar) [inline internal/resolution/datasource/npm_registry_test.go:19]: This is practically the same as the mock http server in the maven tests, and it's probably going to be used in other tests.
I'm going to add a more general mock http server to testutility in another PR

Comment (cuixq) [reply internal/resolution/datasource/npm_registry_test.go:19]: that would be nice! I definitely implement this again for Maven parent support....

File: internal/resolution/datasource/npm_registry_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	if _, err := w.Write(resp); err != nil {
+		log.Fatalf("Write: %v", err)
+	}
+}
+
+func TestNpmRegistryClient(t *testing.T) {
+	t.Parallel()
+
+	//nolint:gosec  // "Potential hardcoded credentials" :)
+	const (
```
Comment (cuixq) [inline internal/resolution/datasource/npm_registry_test.go:67]: how are these credentials generated?

**Review by cuixq — APPROVED** [review]

File: internal/resolution/datasource/npm_registry_test.go (hunk 2 above)
Comment (michaelkedar) [reply internal/resolution/datasource/npm_registry_test.go:67]: These are just random base64-encoded strings. `auth` is supposed to be the encoded `username:password`, but I'm really just checking these are included in the Authorization header.

These I just randomly went with `cool:auth` and `nice-token` :slightly_smiling_face:

**Review by another-rex — APPROVED** [review]

**Pull request merged**